tmp/parameters.txt
# ----------------------------------------------
# Parameter file for map-matching               
# ----------------------------------------------
# User : marie-dominique
# Date : 13/07/2022 13:20:03
# ----------------------------------------------

# Input files
input.network.path = "/home/marie-dominique/CHOUCAS/MAPMATCHER/topo_l93_2d.csv"
input.track.path = "/home/marie-dominique/CHOUCAS/MAPMATCHER/r843.csv"

# Network format
network.header = true
network.delimiter = ","
network.geom.name = ""wkt""
network.source.name = ""source""
network.target.name = ""target""
network.edge.name = ""link_id""
network.oneway.name = ""

# Track format 
track.header = false
track.delimiter = ","
track.columns.x.id = 2
track.columns.y.id = 3
track.columns.t.id = -1
track.error.code = "-1"
track.date.fmt = "yyyy-mm-dd hh:mm:ss"

# Computation parameters 
computation.sigma = 4.0
computation.beta = 0.2
computation.radius = 35.0
computation.transition = 0.0
computation.limit.speed = 1.7976931348623157E308
computation.autocorrelation = 60.0
computation.scope = 100.0
computation.angle = 0.0
computation.distribution = normal

# Additional options 
option.max.candidates = -1
option.failure.skip = true
option.confidence.ratio = false
option.projection = false
option.precompute.distances = true
option.buffer.type = full_network
option.buffer.radius = 300.0
option.make.topology = false
option.sort.nodes = false
option.simplify = false
option.ref.network = false
bias.x = 0.0
bias.y = 0.0
topology.tolerance = 0.01
network.rmse = 0.0
confidence.ratio = 0.0
network.inaccuracies = false

# Output 
output.path = "/home/marie-dominique/CHOUCAS/MAPMATCHER/res4/"
output.clear = true
output.report = true
output.graphics = false
output.debug = false
output.suffix = "_mm.dat"
output.delimiter = ","
output.mute = false
output.errors = true
output.parameters = true
output.index = true
output.rmse = true
output.abs.type = from_source_m
output.confidence = false
output.index.all.edge = false
output.index.coords = false
output.index.format.csv = true
output.rmse.type = after



util/parameters.py
# -*- coding: utf-8 -*-

from datetime import datetime
import os
from pathlib import Path


def getResultPath(resultatpath):
    return resultatpath + '/'


def getGpsMMPath(resultatpath, gpsPath):
    chemin = getResultPath(resultatpath)
    gpsmmpath = chemin + Path(gpsPath).stem + "_mm.dat"
    return gpsmmpath


def getNetworkMMPath(resultatpath):
    chemin = getResultPath(resultatpath)
    networkmmpath = chemin + "network_topo.wkt"
    return networkmmpath


def getTronconTriePath(resultatpath):
    chemin = getResultPath(resultatpath)
    networkmmpath = chemin + "troncon_track_trie.csv"
    return networkmmpath


def createParamFile(resultatpath, networkpath, gpspath,radius):
    
    now = datetime.now()
    dt_string = now.strftime("%d/%m/%Y %H:%M:%S")
    
    fp = open(os.path.dirname(__file__) + "/../tmp/parameters.txt", "w")
    fp.write("# ----------------------------------------------\n")
    fp.write("# Parameter file for map-matching               \n")
    fp.write("# ----------------------------------------------\n")
    fp.write("# User : " + os.getlogin() + "\n")
    fp.write("# Date : " + dt_string + "\n")
    fp.write("# ----------------------------------------------\n")
    fp.write("\n")
    
    fp.write("# Input files\n")
    fp.write('input.network.path = "' + networkpath.strip() + '"\n')
    fp.write('input.track.path = "' + gpspath + '"\n')
    fp.write("\n")
    
    fp.write("# Network format\n")
    fp.write("network.header = true\n")
    fp.write('network.delimiter = ","\n')
    fp.write('network.geom.name = ""wkt""\n')
    fp.write('network.source.name = ""source""\n')
    fp.write('network.target.name = ""target""\n')
    fp.write('network.edge.name = ""link_id""\n')
    fp.write('network.oneway.name = ""\n')
    fp.write("\n")
    
    fp.write('# Track format \n')
    fp.write('track.header = false\n')
    fp.write('track.delimiter = ","\n')
    fp.write('track.columns.x.id = 2\n')
    fp.write('track.columns.y.id = 3\n')
    fp.write('track.columns.t.id = -1\n')
    fp.write('track.error.code = "-1"\n')
    fp.write('track.date.fmt = "yyyy-mm-dd hh:mm:ss" \n')
    fp.write("\n")
    
    fp.write('# Computation parameters \n')
    fp.write('computation.sigma = 4.0                \n')
    fp.write('computation.beta = 0.2                 \n')
    fp.write('computation.radius = ' + str(radius) + '\n')
    fp.write('computation.transition = 0.0           \n')
    fp.write('computation.limit.speed = 1.7976931348623157E308\n')
    fp.write('computation.autocorrelation = 60.0     \n')
    fp.write('computation.scope = 100.0              \n')
    fp.write('computation.angle = 0.0                \n')
    fp.write('computation.distribution = normal      \n')
    fp.write("\n")

    fp.write('# Additional options \n')
    fp.write('option.max.candidates = -1             \n')
    fp.write('option.failure.skip = true             \n')
    fp.write('option.confidence.ratio = false        \n')
    fp.write('option.projection = false              \n')
    fp.write('option.precompute.distances = true     \n')
    fp.write('option.buffer.type = full_network      \n')
    fp.write('option.buffer.radius = 300.0           \n')
    fp.write('option.make.topology = false           \n')
    fp.write('option.sort.nodes = false              \n')
    fp.write('option.simplify = false                \n')
    fp.write('option.ref.network = false             \n')
    fp.write('bias.x = 0.0                           \n')
    fp.write('bias.y = 0.0                           \n')
    fp.write('topology.tolerance = 0.01              \n')
    fp.write('network.rmse = 0.0                     \n')
    fp.write('confidence.ratio = 0.0                 \n')
    fp.write('network.inaccuracies = false           \n')
    fp.write("\n")

    fp.write('# Output \n')
    fp.write('output.path = "' + getResultPath(resultatpath) + '"\n')
    fp.write('output.clear = true\n')
    fp.write('output.report = true\n')
    fp.write('output.graphics = false\n')
    fp.write('output.debug = false\n')
    fp.write('output.suffix = "_mm.dat"\n')
    fp.write('output.delimiter = ","\n')
    fp.write('output.mute = false\n')
    fp.write('output.errors = true\n')
    fp.write('output.parameters = true\n')
    fp.write('output.index = true\n')
    fp.write('output.rmse = true\n')
    fp.write('output.abs.type = from_source_m\n')
    fp.write('output.confidence = false\n')
    fp.write('output.index.all.edge = false\n')
    fp.write('output.index.coords = false\n')
    fp.write('output.index.format.csv = true\n')
    fp.write('output.rmse.type = after\n')
    fp.write("\n")
    fp.close()

    return (fp, os.path.realpath(fp.name))


